### ISSUE TITLE
about temperature_mean

### ISSUE THREAD

**Issue by lipingcoding** [frank-xwang/RIDE-LongTailRecognition#5]
I found there is a $temperature_mean * temperature_mean$ ahead of kl-divergence loss, which isn't include in paper, what's the purpose of it?

**Comment by TonyLianLong** [comment]
Thanks for your question about this. Sorry for the confusion that we did not explain clearly in the first version of our report.

The $temperature_mean$ is a variable that remains constant during training. It's a scalar summarizing the temperature: temperature's mean. The reason for having this temperature is similar to knowledge distillation that we need to regulate the gradient by multiplying the temperature outside of KL div twice (for the exact formula for KL div multiplication, see https://arxiv.org/abs/1503.02531). It is true that we did not have it explicitly in the paper now, but since the temperature is a constant in training, it is also viewed as a part of lambda, which is a hyperparam and also a constant, and viewed as a way to make tuning lambda easier. Therefore, the formula is technically correct, even though it does not explicitly state this variable. We removed the temperature_mean in the equation in paper to avoid unnecessary and potentially confusing symbols, but it seems that we still have some ways to make it clearer. We are planning to revise the paper soon to provide more details to clear questions and discrepancies like this and give more explanations for implicit assumptions. At that time, we will post the update on the project website and arxiv and let you know. I hope this answers your question.
